vmdb/db/fixtures/ae_datastore/ManageIQ/Cloud/VM/Provisioning/Placement.class/__methods__/best_fit_amazon.rb

```diff
@@ -15,3 +15,43 @@
 #  best Availability Zone for you based on system health and available capacity. If you launch additional
 #  instances, only specify an Availability Zone if your new instances must be close to, or separated from,
 #  your running instances."
+#
+#
+# VPC & Subnet
+# For C4 and T2 instances we have to provide VPC and Subnet.
+# http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html
+# Read section marked "EC2-VPC-only Support"
+# For other instance types EC2 provides the defaults.
+# This method checks the instance types and sets the VPC and subnet for T2 and C4 instances
+#
+
+VPC_INSTANCE_GROUPS = %w(C4 T2)
+
+def set_property(prov, image, list_method, property)
+ if prov.get_option(property.to_sym).nil?
+    result = prov.send(list_method)
+    $evm.log("debug", "#{property} #{result.inspect}")
+    object = result.first
+
+    if object
+      prov.send("set_#{property}".to_sym, object)
+      $evm.log("info", "Image=[#{image.name}] #{property}=[#{object.name}]")
+    end
+  end
+end
+
+# Get variables
+prov     = $evm.root["miq_provision"]
+image    = prov.vm_template
+raise "Image not specified" if image.nil?
+
+instance_type  = prov.get_option_last(:instance_type).split(':').first
+instance_group = instance_type.split('.').first.upcase
+
+if VPC_INSTANCE_GROUPS.include?(instance_group)
+  $evm.log("info", "Setting VPC parameters for instance type =[#{instance_type}]")
+  set_property(prov, image, :eligible_cloud_networks, 'cloud_network')
+  set_property(prov, image, :eligible_cloud_subnets,  'cloud_subnet')
+else
+  $evm.log("info", "Using EC2 for default placement of instance type  =[#{instance_type}]")
```

Comment:
You have 2 spaces before the equal sign here which is inconsistent with the other logging where you use no spaces.